service/controller/v25/resource/tccp/update.go

```diff
@@ -229,40 +229,12 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 // changes. In case anything else changes as well, scaling is not allowed, since
 // any other changes should be covered by general updates, which is a separate
 // step.
-func (r *Resource) shouldScale(ctx context.Context, cr v1alpha1.AWSConfig, currentState, desiredState StackState) (bool, error) {
+func (r *Resource) shouldScale(ctx context.Context, cr v1alpha1.AWSConfig) (bool, error) {
 	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
 
-	if currentState.MasterImageID != desiredState.MasterImageID {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "not scaling due to master image id")
-		return false, nil
-	}
```

Comment:
I think we carried all of that for legacy reasons. We return `false` by default. So we only need to consider the enabling conditions, which are kept.

Reply:
I don't quite understand removing this?
IMO this gives us some better logs and if they were added then I assume that there are cases where they are needed to debug easier.

Reply:
It might be I added them when this code got written first. The mechanism is clear though and the code unnecessarily complex. It does not make sense to log all kinds of reasons why a scaling action might not occur, because that can have all kinds of reasons. In case the system behaves unexpectedly we have to debug further. IMO there is no need to keep the legacy code though, as we quite evolved since.